internal/conduit/version.go

```
@@ -0,0 +1,107 @@
+package conduit
+
+import (
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+type Flags struct {
+	args *Args
+}
+
+type Args struct {
+	PipelineFile   string
+	ConnectorsPath string
+	DBPath         string
+	ProcessorsPath string
+}
+
+func NewFlags(fns ...func(*Args)) *Flags {
+	var args Args
+	for _, fn := range fns {
+		fn(&args)
+	}
+	return &Flags{args: &args}
+}
+
+func (f *Flags) ForVersion(ver string) []string {
+	constraints := map[string]string{
+		"v011": "< 0.12.0",
+		"v012": ">= 0.12.0, < 0.13.0",
+	}
+
+	sanitized, _ := strings.CutPrefix(ver, "v")
+	v, _ := semver.NewVersion(sanitized)
+
+	for key, rule := range constraints {
```

Small nitpick, if you change the key to the be the constraint, `map[semver.Constraint]func()[]string)`


you can essentially loop over 

```golang
for rule, fn := range ... {
   if rule.Check(ver) {
      return fn() 
   }
}
```

---

That strategy would throw a parse error, since I would still need to ignore the error that may be returned from NewConstraint. I could declare the constraints before making the map, but that wouldn't save lines

---

Yea that is a good point, though the errs returned by parsing new version and constraints should be checked, anyway. Referring to NewVersion and NewConstraint, this will reduce the errors with typos in code.

You can setup the constraints in `NewFlags` and make them part of `Flags`  and yea that will be a bit more work.